src/main/scala/zio/s3/S3.scala

```diff
@@ -0,0 +1,395 @@
+                     _.createMultipartUpload(
+                       CreateMultipartUploadRequest
+                         .builder()
+                         .bucket(bucketName)
+                         .key(key)
+                         .contentType(contentType)
+                         .build()
+                     )
+                   ).map(_.uploadId())
+
+        parts <- content
+                //part size limit is 5Mb, required by amazon api
+                  .chunkN(5 * 1024 * 1024)
+                  .chunks
+                  .zipWithIndex
+                  .mapMPar(n) {
+                    case (chunk, partNumber) =>
+                      execute(
+                        _.uploadPart(
+                          UploadPartRequest
+                            .builder()
+                            .bucket(bucketName)
+                            .key(key)
+                            .partNumber(partNumber + 1)
+                            .uploadId(uploadId)
+                            .contentLength(chunk.length.toLong)
+                            .build(),
+                          AsyncRequestBody.fromBytes(chunk.toArray)
+                        )
+                      ).map(r => CompletedPart.builder().partNumber(partNumber + 1).eTag(r.eTag()).build())
+                  }
+                  .runCollect
+                  .mapError(S3ExceptionLike)
+
+        _ <- execute(
+              _.completeMultipartUpload(
+                CompleteMultipartUploadRequest
+                  .builder()
+                  .bucket(bucketName)
+                  .key(key)
+                  .multipartUpload(CompletedMultipartUpload.builder().parts(parts.asJavaCollection).build())
+                  .uploadId(uploadId)
+                  .build()
+              )
+            )
+      } yield ()
+
+    def execute[T](f: S3AsyncClient => CompletableFuture[T]): ZIO[Any, S3Exception, T] =
+      fromCompletionStage(UIO.apply(f(unsafeClient))).refineToOrDie[S3Exception]
+  }
+
+  private[s3] case class S3ExceptionLike(error: Throwable)
+      extends S3Exception(S3Exception.builder().message(error.getMessage).cause(error.getCause))
+
+  type StreamResponse = ZStream[Any, Throwable, Chunk[Byte]]
+
+  final private[s3] case class StreamAsyncResponseTransformer(cf: CompletableFuture[StreamResponse])
+      extends AsyncResponseTransformer[GetObjectResponse, StreamResponse] {
+    override def prepare(): CompletableFuture[StreamResponse] = cf
+
+    override def onResponse(response: GetObjectResponse): Unit = ()
+
+    override def onStream(publisher: SdkPublisher[ByteBuffer]): Unit = {
+      cf.complete(publisher.toStream().map(ChunkUtils.fromByteBuffer))
+      ()
+    }
+
+    override def exceptionOccurred(error: Throwable): Unit = {
+      cf.completeExceptionally(error)
+      ()
+    }
+  }
+
+  object Live {
+
+    def connect(region: Region, credentials: S3Credentials): Managed[ConnectionError, S3] =
```

Comment:
```suggestion
    def connect(region: Region, credentials: S3Credentials): Layer[ConnectionError, S3] =
```

Move outside of `Live` and just put inside companion object of S3.